pages/index.tsx

```diff
@@ -102,98 +210,134 @@ export default function Index() {
+                  <MenuIcon />
+                </Button>
+              </Grid>
+              <Grid item>
+                <Typography style={{ marginTop: `8px` }}>
+                  <strong>Current page:</strong> {selectedMenuItem}
+                </Typography>
+              </Grid>
+            </Grid>
+          </Box>
+          <Menu
+            anchorEl={anchorEl}
+            keepMounted
+            open={Boolean(anchorEl)}
+            onClose={handleClose}
+          >
+            {options.map((option, index) => (
+              <MenuItem
+                key={index}
+                selected={index === options.indexOf(selectedMenuItem)}
+                onClick={(_) => handleMenuItemClick(index)}
+              >
+                {option}
+              </MenuItem>
+            ))}
+          </Menu>
+        </div>
+      </Hidden>
+      {selectedMenuItem === "General Info" && (
+        <>
+          <Blurb>
+            <Typography>
+              The Expiring Multi Party (EMP) is{" "}
+              <a
+                href="https://umaproject.org/"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                UMA
+              </a>
+              's most current financial smart contract template. This UI is a
+              community-made tool to make interfacing with the protocol easier,
+              please use at your own risk. The source code can be viewed{" "}
+              <a
+                href="https://github.com/UMAprotocol/emp-tools"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                here
+              </a>
+              . UMA's main Github can be viewed{" "}
+              <a
+                href="https://github.com/UMAprotocol/protocol"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                here
+              </a>
+              .
+            </Typography>
+          </Blurb>
+          <EmpState />
+        </>
+      )}
+      {selectedMenuItem === "Manage Position" && <ManagePosition />}
+      {selectedMenuItem === "All Positions" && <AllPositions />}
+      {selectedMenuItem === "yUSD Yield" && <Yield />}
+      {selectedMenuItem === "Wrap/Unwrap WETH" && <Weth />}
+      {selectedMenuItem === "Analytics" && <Analytics />}
+    </>
+  );
+}
+
+function NoContractPage() {
+  const { signer } = Connection.useContainer();
+  return (
+    <Blurb>
+      {signer ? (
+        <Typography>
+          Please Select an UMA contract from the list above.
+        </Typography>
+      ) : (
+        <Typography>Please connect your wallet before continuing.</Typography>
+      )}
+    </Blurb>
+  );
+}
+
+export default function Index() {
+  const { contract, isValid } = SelectedContract.useContainer();
+  // Use callback here will prevent re mounting components unless contract changes
+  const selectPage = useCallback(() => {
```

Comment:
This index component now switches between 3 states. Perp contract selcted, Emp contract selected, and no/invalid contract selected.